tests/queries/0_stateless/03165_string_functions_with_token_text_indexes.sql

```diff
@@ -195,3 +197,124 @@ FROM (
 WHERE explain LIKE '%Parts:%';
 
 SELECT * FROM 03165_token_ft WHERE match(message, ' xyz ');
+
+SELECT '';
+SELECT '-- No skip for like with matching substring';
+
+SELECT trim(explain)
+FROM (
+    EXPLAIN indexes = 1 SELECT * FROM 03165_token_ft WHERE like(message, '%rvice is not rea%')
+)
+WHERE explain LIKE '%Parts:%';
+
+SELECT * FROM 03165_token_ft WHERE like(message, '%rvice is not rea%');
+
+SELECT '';
+SELECT '-- Skip for like with non-matching string';
+
+SELECT trim(explain)
+FROM (
+    EXPLAIN indexes = 1 SELECT * FROM 03165_token_ft WHERE like(message, '%foo%')
```

Comment:
(The fact that we skip here confuses me (I am not saying that this is wrong behavior).

`foo` could be a substring within a token. Example:
- Indexed document: `Hello World`
- Search filter: `col LIKE '%ll%'`

With the splitByNonAlpha tokenizer, the index will store terms `Hello` and `World`. To answer the search filter using the text index, we'll need to do a full scan of the text index dictionary. Such logic is currently not implemented.)

Reply:
No, you're right, and you caught something I missed. Functions such as like should also use the old code logic, I've updated it in my latest commit.